rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,521 @@
+
+                event = HistoryBaseEntry(
+                    event_identifier=transaction.tx_hash,
+                    timestamp=ts_sec_to_ms(transaction.timestamp),
+                    sequence_index=self.base_tools.get_next_sequence_counter(),
+                    location=Location.BLOCKCHAIN,
+                    asset=self.peth,
+                    balance=Balance(amount=amount),
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.SPEND,
+                    location_label=liquidator,
+                    notes=f'Liquidate {amount} PETH for CDP {cdp_id}',
+                    counterparty=CPT_SAI,
+                )
+                already_decoded = True
+                return event, []
+
+        return None, []
+
+    def _decode_makerdao_deposit_weth_eth_event(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        owner_address = hex_or_bytes_to_address(tx_log.topics[1])
+        amount_raw = hex_or_bytes_to_int(tx_log.topics[2])
+        amount = asset_normalized_value(amount_raw, self.weth)
+
+        # check if ETH is sent to sai proxy contract or a proxy and mark it
+        # as a deposit since the proxy handles the conversion to weth
+        # and transfer to the sai tub contract
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount.is_close(amount, max_diff='0.005') is True
+            ):
+                event.balance.amount = amount
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {self.eth.symbol} for PETH in SAI vault'  # noqa: E501
+                event.counterparty = CPT_SAI
+                break
+
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.counterparty == MAKERDAO_SAI_PROXY_CONTRACT and
+                event.asset == self.eth
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {self.eth.symbol} for PETH in SAI vault'  # noqa: E501
+                event.counterparty = CPT_SAI
+                break
+
+        # NotERC20Conformant is never raised because PETH is a valid ERC20 token
+        get_or_create_evm_token(
```

Comment:
Why not do that at the init and use the result of the function? Now you are using it at every single point and time this function gets called